empyrical/utils.py

```diff
@@ -205,14 +224,39 @@ def ensure_directory(path):
             raise
 
 
+def compute_returns(prices):
+    """
+    Computes correctly-indexed returns from prices.
+
+    Parameters
+    ----------
+    prices : pd.Series or pd.DataFrame
+        Prices of assets in wide-format, with assets as columns,
+        and indexed by datetimes.
+
+    Returns
+    -------
+    returns : pd.Series or pd.DataFrame
+        Returns of assets in wide-format, with assets as columns,
+        and index coerced to be tz-aware.
+    """
+
+    rets = prices.pct_change().dropna()
```

Comment:
Calling `dropna()` and modifying the index here feel philosophically out of place to me for empyrical. 

In my mind at least, the primary purpose of empyrical is to be low-level infrastructure that can be shared betwen Zipline, PyFolio, and Alphalens. Zipline pretty much never wants to call a function that silently discards bad data, or that changes the localization of a datetime index, because both of those behaviors prevent us from noticing bugs, and they incur nontrivial performance overhead.

Calling `dropna()` in particular here seems a bit fraught if we expect DataFrame input, because the behavior there is that pandas will drop an entire row if there are any NaNs in the row. If we expect the user to have missing data, then that seems like not a great behavior (if they pass enough columns, it seems likely that they'll drop a large percentage of their input data). If we don't expect them to have missing data, then we're just incurring extra cost for no reason.

If we want to include a simple returns function in empyrical, I would probably put it in `stats` and write it as something like:

```python
def simple_returns(prices):
    """Compute simple returns from a timeseries of prices.

    <rest of the docstring>
    """
    if isinstance(prices, (pd.DataFrame, pd.Series)):
        return prices.pct_change().iloc[1:]
    else:
        # Assume ndarray
        out = np.diff(prices, axis=0)
        np.divide(out, prices[:-1], out=out)
        return out
```

Reply:
This also needs a test if we decide to keep it.

Reply:
This will sound bad but I was calling `dropna` because that was the way it was done before, and I assumed there was a reason for that 😦 

I agree that this function belongs in `stats`, though. The question is whether or not we want `empyrical` to tz-localize anything. Philosophically, I feel as if that should be the responsibility of downstream packages like alphalens and pyfolio, since empyrical is only responsible for low-level financial calculations, and is not meant to be user-facing at all. This relates to the input sanitization PR for pyfolio.

@twiecki thoughts?

Reply:
Yes, `.iloc[1:]` is a bit clearer on the intent.

Reply:
And yeah, I'd be happy if we just completely did away with the whole tz-aware thing. Since we don't load any data in the future, we can probably just ignore it and the user has to ensure that it matches up.